internal/storage/rocksdb/store.go

```diff
@@ -531,6 +618,15 @@ func (rdb *rocksDB) getMultipleKeys(ro *gorocksdb.ReadOptions, keys [][]byte) ([
 	for i, value := range values {
 		if value != nil {
 			key, val := keys[i], toByteArray(value)
+			//check ttl.
+			expireTs, _ := getExpireTs(val)
+			if expireTs > 0 && expireTs < time.Now().Unix() {
+				//triger delete.
+				rdb.Delete(key)
+				continue
```

Comment:
Two issues:
1. we must add `nil` to the `results` against the current key as that's the expected semantics for missing keys in RocksDB
2. `value` must be `Free`d to avoid memory leak

Reply:
Fixed